daemon/src/patch.rs

```diff
@@ -0,0 +1,169 @@
+//! [`list`] all the [`Patch`]es for project.
+
+use either::Either;
+use librad::{
+    git::types::{namespace::Namespace, Reference, RefsCategory},
+    peer::PeerId,
+    signer::BoxedSigner,
+};
+use radicle_git_ext::RefspecPattern;
+
+use crate::{
+    project::{self, peer},
+    state, Person,
+};
+
+const TAG_PREFIX: &str = "radicle-patch/";
```

Comment:
Adding a `pub fn rad_patch` to `impl`s [here](https://github.com/radicle-dev/radicle-link/blob/master/librad/src/git/types/reference.rs#L160) would help with interacting with these. 

Bikeshedding moment: the path could be `refs/namespaces/<ns>/refs/rad/patches`

Reply:
> Bikeshedding moment: the path could be `refs/namespaces/<ns>/refs/rad/patches`

If patches don’t live under the `tags` ref category then pushing will be slightly more complicated: `git push tags/rad/patches/*:rad/patches/*` instead of `git push --tags`. The upside is that we’re not messing with the regular `tags` category.

I have no strong preference so I’m happy to change this.

Reply:
Ah yes, this is a good point :) 

> `git push tags/rad/patches/*:rad/patches/*`

I wonder if we could extend the `include` machinery to do this for us :thinking:

Reply:
> I wonder if we could extend the `include` machinery to do this for us thinking

Are you thinking about leveraging the [`remote.*.push` option](https://git-scm.com/docs/git-config#Documentation/git-config.txt-remoteltnamegtpush)? This might be dangerous because setting it messes with the users workflow. `git push rad` (without any refspecs) will use the configured value instead of pushing the current branch. Especially if we do this on checkout in the background users will not be aware.

Reply:
Well currently for pushing branches we have:

```
[remote "rad"]
	url = rad://hnrkqyoojqyxtm8kcwokr6r4rfc3ey18xyeho.git
	fetch = +refs/heads/*:refs/remotes/rad/*
```

I thought it would have been possible to do the following:
```
[remote "rad"]
	url = rad://hnrkqyoojqyxtm8kcwokr6r4rfc3ey18xyeho.git
	fetch = +refs/heads/*:refs/remotes/rad/*
	fetch = +refs/rad/patches/*:refs/tags/*
```

However, after testing it, it just sends them to `refs/tags` in the monorepo regardles :/ Likely `git` Doing What I Mean :upside_down_face: